### DISCUSSION TITLE
Trying to decipher how this script set title of video in mpv-android

### DISCUSSION THREAD

**Discussion by tathastu871** [pystardust/ani-cli#1031] [Q&A] [answered]
Can you explain how this script sets title in video in mpv-android

i tried all means but unable to do it in my script

**Comment by CoolnsX** [comment, marked as answer]
Tbh we don't know too.. and we are unable to find a way to set title in mpv.. vlc-android supports that but afaik mpv-android doesn't

**Reply by tathastu871** [reply]
i created a script that set custom title to mpv-android but it extremely slows down video startup by few seconds to 1 minute. Works for both urls and local videos so far as long as any update to mpv does not break my script

However my method not working for some  urls
![Screenshot_20230212-192021_mpv](https://user-images.githubusercontent.com/65007212/218315079-0781022a-38ba-4645-b37b-2781346de2fd.jpg)

**Reply by CoolnsX** [reply]
Can u share the script?

**Reply by port19x** [reply]
> i created a script that set custom title to mpv-android but it extremely slows down video startup by few seconds to 1 minute. Works for both urls and local videos so far as long as any update to mpv does not break my script
> 
> However my method not working for some urls ![Screenshot_20230212-192021_mpv](https://user-images.githubusercontent.com/65007212/218315079-0781022a-38ba-4645-b37b-2781346de2fd.jpg)

Holy crap, up to 1 minute?
Share it anyway, would love to have this upstream

**Reply by Veroniclover** [reply]
I found this [one](https://github.com/Rendevior/mpv-android-argparse), it works well with ani-cli.
![screen](https://user-images.githubusercontent.com/73184907/219041195-4b1758e7-1ab8-4466-892a-adb00f69df26.jpg)

**Reply by tathastu871** [reply]
works some time. already tried that and collaborated with author.

**Comment by tathastu871** [comment]
add to mpv-android > setting > advance > edit mpv.conf
add 
include="/storage/emulated/0/mpv.conf"

we can make wrapper like below and add to $PREFIX/etc/bash.bashrc in termux
```
mpv() { 
if [[ "$1" =~ "--media-title" ]]; then
title="${1#--}"
shift
fi
stream="$1"
rm /storage/emulated/0/mpv.conf; echo "$title" >> /storage/emulated/0/mpv.conf
am start -n is.xyz.mpv/.MPVActivity -a android.intent.action.VIEW -d "$stream"
rm /sdcard/mpv.conf
}
```
Try: 
```
mpv --media-title="Cutom_Title" "file:///sdcard/file.mp4"
mpv --media-title="Cutom_Title" "Stream Url"
```
**Note: if --media-title not working replace it with --force-media-title**

This is a workaround that can be used to set title in mpv-android player
